src/relay/pass/pattern_util.h

```diff
@@ -192,6 +192,21 @@ inline Constant MakeConstantScalar(DataType dtype, T value) {
   return ConstantNode::make(arr);
 }
 
+/*!
+ * \brief Check if two expressions are equal scalars.
+ * \param a The expression to be checked.
+ * \param b The expression to be checked
+ * \return Whether two expressions are equal scalars.
+ */
+inline bool IsEqualScalar(const Expr& a, const Expr& b) {
+  const auto* constant_a = a.as<ConstantNode>();
```

Comment:
constant_a is nullptr with `relay.var("x", shape=(1, 16))` in your test script.  Is this what you expect?

Reply:
yes, this function is intended to enable combining different Constant instance with the same value